Python/optimizer.c

```diff
@@ -0,0 +1,245 @@
+
+#include "Python.h"
+#include "opcode.h"
+#include "pycore_interp.h"
+#include "pycore_opcode.h"
+#include "pycore_pystate.h"
+#include "cpython/optimizer.h"
+#include <stdbool.h>
+#include <stdint.h>
+#include <stddef.h>
+
+static int32_t
+get_next_free_in_executor_array(PyCodeObject *code)
+{
+    _PyExecutorArray *old = code->co_executors;
+    int size = 0;
+    int capacity = 0;
+    if (old != NULL) {
+        size = old->size;
+        capacity = old->capacity;
+        if (capacity >= 256) {
+            PyErr_Format(PyExc_SystemError,
+                        "too many executors for code object");
```

Comment:
Should this really be an exception? Is it a bug to have a code object with >255 backward jumps in it? Shouldn't we instead just silently refuse to add any more executors if we reach this limit?

Reply:
Yes. In my original  design, this could be caught by the optimizer, so at least it would know there was an issue.

Functions with more than 256 backedges do seem unlikely, so I guess just giving up would be fine. 
I'll do that for now. We can always revisit this later.